### PR TITLE
feat: update some recipes and fix some deprecations

### PR DESCRIPTION
**Pull request by JellyBellyDev**
[enhancement/update-recipes aaa04a5] feat: update some recipes and fix some deprecations

Please help me to refactoring `BadgeSubscriberTest`.  
Now the class "Symfony\Component\HttpKernel\Event\ExceptionEvent" is declared "final" and cannot be mocked.

I found this: https://github.com/symfony/symfony/issues/38627

### CODE REVIEW

File: public/index.php [hunk 1]
```diff
@@ -1,28 +1,22 @@
...
 }
 
-if ($trustedProxies = $_SERVER['TRUSTED_PROXIES'] ?? $_ENV['TRUSTED_PROXIES'] ?? false) {
-    $trustedProxies = array_map('trim', explode(',', $trustedProxies));
-    Request::setTrustedProxies($trustedProxies, Request::HEADER_X_FORWARDED_AWS_ELB);
-}
-
-if ($trustedHosts = $_SERVER['TRUSTED_HOSTS'] ?? $_ENV['TRUSTED_HOSTS'] ?? false) {
-    Request::setTrustedHosts([$trustedHosts]);
-}
```
Comment (JellyBellyDev) [inline public/index.php:22]: @fabiocicerchia perhaps this must remain? Are we behind a proxy in production?

**Comment by AlessandroMinoccheri** [comment]
IMHO @JellyBellyDev you can create a new instance of ExceptionEvent passing mocked object. Something like this>

`new ExceptionEvent($httpClientInterfaceMock->reveal(), $request->reveal(), $requestType, $throwable);
`
In this way, tests will be a little bit more complicated but you can mock arguments.
What do you think?

**Comment by JellyBellyDev** [comment]
> IMHO @JellyBellyDev you can create a new instance of ExceptionEvent passing mocked object. Something like this>
> 
> `$exceptionEvent = new ExceptionEvent($httpClientInterfaceMock->reveal(), $request->reveal(), $requestType, $throwable); `
> In this way, tests will be a little bit more complicated but you can mock arguments.
> What do you think?

@AlessandroMinoccheri yes, it seems like the right approach, but I don't know how to refactor the test under ... ideas?

**Comment by AlessandroMinoccheri** [comment]
@JellyBellyDev  something like this

```
$kernel = $this->getMockBuilder(HttpClientInterface::class)
            ->disableOriginalConstructor()
            ->getMock();
        
        $request = $this->getMockBuilder(Request::class)
            ->disableOriginalConstructor()
            ->getMock();

        $requestType = HttpKernelInterface::MASTER_REQUEST; 
        
        $e = $this->getMockBuilder(\Throwable::class)
            ->disableOriginalConstructor()
            ->getMock();
        
        $this->event = new ExceptionEvent(
            $kernel->reveal(),
            $request->reveal(),
            $requestType,
            $e->reveal()
        );
```

This means that we need to mock many methods and I'll suggest starting with this and add step by step all necessary methods.
It can be a lot of work I know but this could be the solution.

**Review by fabiocicerchia — APPROVED** [review]

**Comment by garak** [comment]
> Please help me to refactoring `BadgeSubscriberTest`.
> Now the class "Symfony\Component\HttpKernel\Event\ExceptionEvent" is declared "final" and cannot be mocked.

We can use https://packagist.org/packages/dg/bypass-finals

**Comment by JellyBellyDev** [comment]
close in favor of #501

**Pull request closed without merging**